test/test_jit.py

```diff
@@ -7858,6 +7858,17 @@ def to_str(x):
 
         self.assertEqual("(1, 1)", to_str(1))
 
+    def test_int_cast(self):
+        @torch.jit.script
+        def to_int(x):
+            # type: (str) -> int
+            return int(x)
+
+        self.assertEqual(5, to_int('5'))
```

Comment:
Can you add some more test cases for things like `"5 "` and the failure cases like `"0x20"` (we leave the `base` arg for a follow up PR)

Reply:
Sure, I have added tests for non-decimal numbers. However, there seems to be a slightly incorrect implementation of `stoll` and `stod` in `c10` which doesn't trim spaces before passing it to the stringstream. This means that even a statement like `float('5.5  ')` throws an error.